### DISCUSSION TITLE
Help with reading DAS files

### DISCUSSION THREAD

**Discussion by josephwinston** [DASDAE/dascore#408] [Q&A]
I have DAS data from [Utah FORGE: Well 16B(78)-32 2023 Neubrex Energy Services Circulation Test Period with Fiber Optics Monitoring](https://gdr.openei.org/submissions/1565) and I don't have any issues reading the data with  [h5py](https://docs.h5py.org/en/stable/index.html)

The fragment for h5py would look like:

```python
     hdf_files = {
        'DTS': "C:/Users/jwinston/work/Data/DAS/Neubrex/Utah Forge 16B(78)-32 - DTS - mmf 2 - evo 0.h5",
    }
    hdf_file = h5py.File(hdf_files['DTS'])
    data_set = hdf_file['data']
    print('DTS')
    print(f"shape: {data_set.shape}")
    print(f"type: {data_set.dtype}")
```

The result is:

```
DTS
shape: (458, 6183)
type: float32
```

However, when I use dascore as follows:

```python

import dascore as dc

path = "C:/Users/jwinston/work/Data/DAS/Neubrex/Utah Forge 16B(78)-32 - DTS - mmf 2 - evo 0.h5"
hdf = dc.io.H5Reader(path)

print(hdf)
hdf.close()

file_format, file_version = dc.get_format(path)

```

This produces the following:

```
C:/Users/jwinston/work/Data/DAS/Neubrex/Utah Forge 16B(78)-32 - DTS - mmf 2 - evo 0.h5 (File) ''
Last modif.: '2024-07-04T16:46:00+00:00'
Object Tree: 
/ (RootGroup) ''
/data (CArray(458, 6183)zlib(4)) ''
/depth (CArray(6183,)zlib(4)) ''
/stamps (CArray(458,)zlib(4)) ''
/stamps_unix (CArray(458,)zlib(4)) ''

Traceback (most recent call last):
  File "C:\Users\jwinston\work\BiaTech\Prototpyes\DAS\Hello\Read-dascore.py", line 9, in <module>
    file_format, file_version = dc.get_format(path)
                                ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jwinston\anaconda3\envs\DAS\Lib\site-packages\dascore\io\core.py", line 886, in get_format
    out = FiberIO.manager._get_format(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jwinston\anaconda3\envs\DAS\Lib\site-packages\dascore\io\core.py", line 364, in _get_format
    raise UnknownFiberFormatError(msg)
dascore.exceptions.UnknownFiberFormatError: Could not determine file format of C:/Users/jwinston/work/Data/DAS/Neubrex/Utah Forge 16B(78)-32 - DTS - mmf 2 - evo 0.h5

```

Obviously, I don't know what I am doing.  I don't know how to select the appropriate HDF5 dataset or navigate through groups.

Where do I go look for examples?

**Comment by d-chambers** [comment]
Hey @josephwinston, it's likely that we don't support this particular flavor of H5 yet, but I it isn't difficult to add support.  I will look into it early next week.

**Reply by josephwinston** [reply]
Anything that I can do to help put together a first PR on this issue?  If you want help, point me to where the changes need to be made.

**Reply by d-chambers** [reply]
You can take a look at the [adding a new format](https://dascore.org/contributing/new_format.html) documentation and see if it's something you want to tackle. I am happy to help along the way or I should have time to look at it in a few days.

**Comment by d-chambers** [comment]
I spent a few minutes looking through the dataset and there really aren't any "standard" das files here. I see some DSS, DTS, and some  [DAS FBE](https://docs.energistics.org/PRODML/PRODML_TOPICS/PRO-DAS-000-064-0-C-sv2000.html) (frequency band extracted) data. 

The DSS and DTS files look straightforward to support, and some of them are even small enough to include directly in our test data suite. We don't yet support any non-DAS files, but I don't see a reason not to support them.  The FBE data will be a bit more challenging but we might be able to do it.

I just wanted to let you know I haven't forgotten about this! Will continue next week.

**Comment by josephwinston** [comment]
Understand Thanks for taking the time to to a look. On Jul 12, 2024, at 7:19 PM, Derrick Chambers ***@***.***> wrote:﻿
I spent a few minutes looking through the dataset and there really aren't any "standard" das files here. I see some DSS, DTS, and some  DAS FBE (frequency band extracted) data.
The DSS and DTS files look straightforward to support, and some of them are even small enough to include directly in our test data suite. We don't yet support any non-DAS files, but I don't see a reason not to support them.  The FBE data will be a bit more challenging but we might be able to do it.
I just wanted to let you know I haven't forgotten about this! Will continue next week.

**Reply by eileenrmartin** [reply]
It looks like @d-chambers was able to push the code for Neubrex h5 support to the main branch, with new relevant code in https://github.com/DASDAE/dascore/tree/master/dascore/io/neubrex. @josephwinston if you update DASCore, does the new version work for you?

**Comment by josephwinston** [comment]
Swamped with other work and will try before Monday On Jul 25, 2024, at 4:28 PM, eileenrmartin ***@***.***> wrote:﻿
It looks like @d-chambers was able to push the code for Neubrex h5 support to the main branch, with new relevant code in https://github.com/DASDAE/dascore/tree/master/dascore/io/neubrex. @josephwinston if you update DASCore, does the new version work for you?

**Reply by josephwinston** [reply]
Long weekend of work but no review yet.

**Comment by josephwinston** [comment]
Looks great.

**Reply by josephwinston** [reply]
Please close.
